docs/how-to-guides/feathr-configuration-and-env.md

```diff
@@ -0,0 +1,114 @@
+---
+layout: default
+title: Configuration, environment varialbles, and store secrets in a secure way
+parent: Feathr How-to Guides
+---
+
+# Configuration and environment varialbles in Feathr
+
+Feathr uses a YAML file and a few environment variables to allow end users to have more flexibility. See the example of the following configurations in [this file](https://github.com/linkedin/feathr/blob/main/feathr_project/feathrcli/data/feathr_user_workspace/feathr_config.yaml).
+
+In that YAML file, it contains the configurations that are used by Feathr. All the configurations can be overwritten by environment variables with concatenation of `__` for different layers of this config file. For example, `feathr_runtime_location` for databricks can be overwritten by setting this environment variable: `SPARK_CONFIG__DATABRICKS__FEATHR_RUNTIME_LOCATION`. For example, you can set it in python:
+
+```python
+os.environ['SPARK_CONFIG__DATABRICKS__FEATHR_RUNTIME_LOCATION'] = "https://azurefeathrstorage.blob.core.windows.net/public/feathr-assembly-LATEST.jar"
+```
+
+or in shell environment:
+
+```bash
+export SPARK_CONFIG__DATABRICKS__FEATHR_RUNTIME_LOCATION=https://azurefeathrstorage.blob.core.windows.net/public/feathr-assembly-LATEST.jar
+```
+
+This allows end users to store the configuration in a secure way, say in Kubernetes secrets, key vault, etc. All the configurations available for end users to configure are listed below.
+
+
+# Default behaviors
+
+Feathr will get the configurations in the following order:
+
+1. If the key is set in the envrionment variable, Feathr will use the value of that environment variable
+2. If it's not set in the environment, then a value is retrieved from from the feathr_config.yaml file with the same config key.
+3. If it's not available in the feathr_config.yaml file, Feathr will try to reterive the value from a key vault service. Currently only Azure Key Vault is supported.
+
+# A list of environment variables that Feathr uses
+| Environment Variable     | Description     |
```

Comment:
How about move require status to a new column, the value can be Required or Required if uses XXX. Then description can only focus on indicate what environment var is used for.

Reply:
Thanks, updated